test/sync/src/androidTest/kotlin/io/realm/kotlin/test/shared/SyncedRealmTests.kt

```diff
@@ -577,6 +578,64 @@ class SyncedRealmTests {
         }
     }
 
+    @Test
+    fun mutableRealmInt_syncsCorrectly() = runBlocking {
+        val user = app.createUserAndLogIn(randomEmail(), "password1234")
+        val id = "id-${Random.nextLong()}"
+        val masterObject = SyncObjectWithAllTypes().apply { _id = id }
+
+        val config = createSyncConfig(
+            user = user,
+            partitionValue = partitionValue,
+            schema = setOf(SyncObjectWithAllTypes::class)
+        )
+
+        // Upload data with default values
+        Realm.open(config).use { realm ->
+            realm.write { copyToRealm(masterObject) }
+            realm.syncSession.uploadAllLocalChanges()
+        }
+
+        // Increment counter twice asynchronously
+        val updates = async {
+            for (i in 0..1) {
+                Realm.open(config).use { realm ->
```

Comment:
I'm not sure this actually tests the interesting case. You open the same Realm twice which means that all changes are done to the same file on disk, I would probably open two different Realms.